File: Assets/Scripts/Module/Invoke/FActions/DelayAction.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace FAction
{
    public class DelayAction : FAction
    {
        public override ActionType ActionType => ActionType.Delay;
        // 延迟时间,单位秒
        public float delayTime;

        float remain = -1f;

        public override void Tick()
        {
            if (remain < 0)
            {
                remain = delayTime;
            }

            remain = remain - Time.deltaTime;
            if (remain <= 0)
            {
                this.IsFinish = true;
            }
        }

        public override void Clear()
        {
            base.Clear();
            delayTime = 0;
            remain = -1;
        }


        public override void Reset()
        {
            remain = -1;
            this.IsFinish = false;
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/FAction/DelayAction.lua
---@class CS.FAction.DelayAction : CS.FAction.FAction
CS.FAction.DelayAction = {}

---@field public CS.FAction.DelayAction.delayTime : CS.System.Single
CS.FAction.DelayAction.delayTime = nil

---@property readonly CS.FAction.DelayAction.ActionType : CS.FAction.ActionType
CS.FAction.DelayAction.ActionType = nil

---@return CS.FAction.DelayAction
function CS.FAction.DelayAction()
end

function CS.FAction.DelayAction:Tick()
end

function CS.FAction.DelayAction:Clear()
end

function CS.FAction.DelayAction:Replay()
end